core/store/tx_manager.go

```diff
@@ -378,6 +389,24 @@ func (txm *EthTxManager) BumpGasUntilSafe(hash common.Hash) (*models.TxReceipt,
 	return nil, Unconfirmed, merr
 }
 
+func (txm *EthTxManager) checkDBForConfirmation(tx *models.Tx) (*models.TxReceipt, AttemptState, error) {
+	later, err := txm.orm.FindLaterConfirmedTx(tx)
```

Comment:
Rather than adding one extra TX per EthTx execution, could we have the TX manager simply record the last confirmed nonce?